api/src/services/data-ingestions/create-service.ts

```diff
@@ -0,0 +1,142 @@
+import Papa from "papaparse";
+import { DateTime } from "luxon";
+import { isNil } from "lodash";
+
+import { db } from "@/data/db-client";
+import {
+  DataIngestion,
+  DataIngestionMapping,
+  DataIngestionSource,
+  DataIngestionSourceNames,
+} from "@/data/models";
+import BaseService from "@/services/base-service";
+
+export class CreateService extends BaseService {
+  constructor(
+    private csvBuffer: Buffer,
+    private source_id: number,
+    private user_id: number
+  ) {
+    super();
+  }
+
+  async perform(): Promise<void> {
+    const csvText = this.csvBuffer.toString("utf-8");
+
+    const source = await this.getSourceOrThrow(this.source_id);
+    const rows = this.parseAndValidateCsv(source, csvText);
```

Comment:
What type of "rows" are these? Are they rows of data ingestion attributes or something else?